clang/lib/CConv/RewriteUtils.cpp

```diff
@@ -1423,6 +1423,55 @@ std::string ArrayBoundsRewriter::getBoundsString(Decl *D, bool Isitype) {
   return BString;
 }
 
+class TypedefRewriter : public clang::RecursiveASTVisitor<TypedefRewriter>
+{
+public:
+  explicit TypedefRewriter(ASTContext *C, ProgramInfo &I,
+                           Rewriter& R, RSet &R2)
+      : Context(C), Info(I), Writer(R), RewriteThese(R2) {}
+  bool VisitTypedefDecl(TypedefDecl *D) {
+    bool foundWild = false;
+    for (auto pair : Info.TypedefVariables) {
+      std::set<ConstraintVariable*> cvs2 = Info.TypedefVariables[pair.first];
+      if (pair.first == D->getNameAsString()) {
+        for (auto cv : cvs2) {
+          if (cv->hasWild(Info.getConstraints().getVariables())) {
+            foundWild = true;
+          }
+        }
+      }
+    }
+    SourceRange sourceRange1 = D->getSourceRange();
+    sourceRange1.setBegin(sourceRange1.getBegin().getLocWithOffset(8));
+    sourceRange1.setEnd(sourceRange1.getBegin().getLocWithOffset(6));
+    std::string text = "";
+
+    if (sourceRange1.getBegin().isValid() && sourceRange1.getEnd().isValid()) {
+      text = Writer.getRewrittenText(sourceRange1);
+    }
+
+    if (!foundWild and !(text == "struct")) {
```

Comment:
Are you trying to see if the type of `D` is a struct? There should be easier and more sound methods instead of searching for "struct" at the beginning of the text.